File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/IChannelCollection.vb
'*******************************************************************************************************
'  IChannelCollection.vb - Channel collection interface
'  Copyright © 2008 - TVA, all rights reserved - Gbtc
'
'  Build Environment: VB.NET, Visual Studio 2008
'  Primary Developer: J. Ritchie Carroll, Operations Data Architecture [TVA]
'      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
'       Phone: 423/751-2827
'       Email: jrcarrol@tva.gov
'
'  Code Modification History:
'  -----------------------------------------------------------------------------------------------------
'  02/18/2005 - J. Ritchie Carroll
'       Initial version of source generated
'
'*******************************************************************************************************

Imports System.Runtime.Serialization

''' <summary>This interface represents a protocol independent representation of a collection of any data type.</summary>
<CLSCompliant(False)> _
Public Interface IChannelCollection(Of T As IChannel)

    Inherits IChannel, IList(Of T), ISerializable

End Interface

File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/IChannelCollection.cs
//*******************************************************************************************************
//  IChannelCollection.cs
//  Copyright © 2009 - TVA, all rights reserved - Gbtc
//
//  Build Environment: C#, Visual Studio 2008
//  Primary Developer: James R Carroll
//      Office: PSO TRAN & REL, CHATTANOOGA - MR BK-C
//       Phone: 423/751-4165
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  02/18/2005 - James R Carroll
//       Generated original version of source code.
//
//*******************************************************************************************************

using System.Collections.Generic;
using System.Runtime.Serialization;

namespace PCS.PhasorProtocols
{
    /// <summary>
    /// Represents a protocol independent interface representation of a collection of any
    /// <see cref="IChannel"/> objects.<br/>
    /// This is the base interface implemented by all collections classes in the phasor
    /// protocols library; it is the root of the collection interface hierarchy.
    /// </summary>
    /// <typeparam name="T">Specific <see cref="IChannel"/> type that the <see cref="IChannelCollection{T}"/> contains.</typeparam>
    public interface IChannelCollection<T> : IChannel, IList<T>, ISerializable where T : IChannel
    {
        // Note that the channel collection interface inherits IChannel hence providing cumulative imaging properties
    }
}
